core/includes/backup/backup.database.inc

```diff
@@ -0,0 +1,350 @@
+      $this->log('Unable to restore from file %file because a %type file can\'t be restored to this database.',
+        array(
+          '%file' => $file->filepath(),
+          '%type' => $file->extension(),
+        ),
+        Backup::LOG_ERROR
+      );
+    }
+    else {
+      // Restore the database.
+      $lines_executed = $this->restoreDatabaseFromFile($file);
+
+      if ($this->settings['verbose']) {
+        $this->log('@count SQL commands executed.', array('@count' => $lines_executed), self::LOG_INFO);
+      }
+    }
+
+    // Pull any logged messages out of the file and up to this class.
+    $file_log = $file->getLog();
+    foreach ($file_log as $log) {
+      call_user_func_array($this->log, $log);
+    }
+    $file->clearLog();
+
+    return $lines_executed;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function postRestore(BackupFile &$file) {
+    parent::postRestore($file);
+
+    // If a compressed copy of this backup exists, delete the uncompressed copy.
+    if ($file->lastExtension() === 'sql') {
+      $uncompressed_path = $file->filePath();
+      $file->pushExtension('gz');
+      if (file_exists($file->filepath())) {
+        file_unmanaged_delete($uncompressed_path);
+      }
+      else {
+        $file->popExtension();
+      }
+    }
+  }
+
+  /**
+   * Get the db connection for the specified db.
+   */
+  protected function getDatabaseConnection() {
+    if (!$this->connection) {
+      $parts = explode(':', $this->getTarget());
+      // One of the predefined databases (set in settings.php)
+      if ($parts[0] == 'db') {
+        $key = empty($parts[1]) ? 'default' : $parts[1];
+        $target = empty($parts[2]) ? 'default' : $parts[2];
+      }
+      // An arbitrary database URL such as mysql://user:pass@localhost:3306/db.
+      // This ability is not used directly by core, which only provides UIs for
+      // backing up and restoring existing database connections.
+      else {
+        $url_parts = parse_url($this->getTarget());
+        // If the url is specified build it into a connection info array.
+        if ($url_parts) {
+          $info = array(
+            'driver' => empty($url_parts['scheme']) ? NULL : $url_parts['scheme'],
+            'host' => empty($url_parts['host']) ? NULL : $url_parts['host'],
+            'port' => empty($url_parts['port']) ? 3306 : $url_parts['port'],
+            'username' => empty($url_parts['user']) ? NULL : $url_parts['user'],
+            'password' => empty($url_parts['pass']) ? NULL : $url_parts['pass'],
+            'database' => empty($url_parts['path']) ? NULL : $url_parts['path'],
+          );
+          $key = uniqid('backdrop_backup_tmp_');
+          $target = 'default';
+          Database::addConnectionInfo($key, $target, $info);
+        }
+        // No database selected. Assume the default.
+        else {
+          $key = $target = 'default';
+        }
+      }
+      if ($target && $key) {
```

Comment:
Seems like $target and $key are always set (strings). But maybe I'm missing something...

Reply:
I think you're right, I'll remove this unnecessary condition.